README.md

```diff
@@ -26,7 +43,185 @@ install](http://hoa-project.net/Source.html).
+editor and administrator.
+
+### Create the ACL
+
+We start by creating all the actors, in separated variables for the sake of
+clarity:
+
+```php
+$groupVisitor       = new Hoa\Acl\Group('group_visitor');
+$groupBuyer         = new Hoa\Acl\Group('group_buyer');
+$groupEditor        = new Hoa\Acl\Group('group_editor');
+$groupAdministrator = new Hoa\Acl\Group('group_administrator');
+
+$userAnonymousVisitor = new Hoa\Acl\User('user_visitor_anonymous');
+$userLoggedVisitor    = new Hoa\Acl\User('user_visitor_logged');
+$userProductEditor    = new Hoa\Acl\User('user_editor_product');
+$userBlogEditor       = new Hoa\Acl\User('user_editor_blog');
+
+$permissionRead  = new Hoa\Acl\Permission('permission_read');
+$permissionWrite = new Hoa\Acl\Permission('permission_write');
+$permissionBuy   = new Hoa\Acl\Permission('permission_buy');
+
+$serviceProduct  = new Hoa\Acl\Service('service_product');
+$serviceBlogPage = new Hoa\Acl\Service('service_blog_page');
+```
+
+Then, we put them together: We create an ACL instance, we add services on users
+and groups, we add users on groups, we add groups inside the ACL instance and
+finally we add permissions on groups.
+
+```php
+// Create an ACL instance.
+$acl = new Hoa\Acl();
+
+// Add services to users and groups.
+// The visitor group shares the product and the blog page services.
+$groupVisitor->addServices([$serviceProduct, $serviceBlogPage]);
+// The buyer group shares the product and the blog page services (reminder:
+// Services are not inherited).
+$groupBuyer->addServices([$serviceProduct, $serviceBlogPage]);
+// The product editor user owns the product service.
+$userProductEditor->addServices([$serviceProduct]);
+// The blog editor user owns the blog page service.
+$userBlogEditor->addServices([$serviceBlogPage]);
+
+// Add users to groups.
+// The visitor group contains one anonymous visitor user.
+$groupVisitor->addUsers([$userAnonymousVisitor]);
+// The buyer group contains one logged visitor user.
+$groupBuyer->addUsers([$userLoggedVisitor]);
+// The editor group contains two users: Product editor and blog editor.
+$groupEditor->addUsers([$userProductEditor, $userBlogEditor]);
+
+// Add groups to the ACL instance.
+$acl->addGroup($groupVisitor);
+// The buy group inherits permissions from the visitor group.
+$acl->addGroup($groupBuyer, [$groupVisitor]);
+$acl->addGroup($groupEditor);
+// The administrator group inherits permissions from the editor group.
+$acl->addGroup($groupAdministrator, [$groupEditor]);
+
+// Add permissions.
+// The visitor group has permission to read.
+$acl->allow($groupVisitor, [$permissionRead]);
+// The buy group has permission to buy.
+$acl->allow($groupBuyer, [$permissionBuy]);
+// The editor group has permission to read and write.
+$acl->allow($groupEditor, [$permissionRead, $permissionWrite])
+```
+
+This is important to keep in mind that users and services are not inherited
+between groups.
```

Comment:
Services cannot be inherited because they are either **owned** by a user (so inheritance does not happen here) or **shared** by a group. In this latter case, the group shares a service but it does not make sense (or it even dangerous) to expose to its children.

Now, about users. I don't have strong arguments except that a user is not something that can be inherited. A user belongs to zero or more groups but this is not because it belongs to one group that it belongs to the children of this group.

I don't know if I am clear?